sessionctx/variable/sysvar.go

```diff
@@ -1128,7 +1128,7 @@ var defaultSysVars = []*SysVar{
 		s.EnableTablePartition = val
 		return nil
 	}},
-	{Scope: ScopeSession, Name: TiDBEnableListTablePartition, Value: Off, Type: TypeBool, skipInit: true, SetSession: func(s *SessionVars, val string) error {
+	{Scope: ScopeGlobal | ScopeSession, Name: TiDBEnableListTablePartition, Value: Off, Type: TypeBool, skipInit: true, SetSession: func(s *SessionVars, val string) error {
```

Comment:
This has `skipInit` set because when I developed this feature I just copied the existing behavior. Because you are adding a global scope now, I don't think this is intentional:

What `skipInit` does is avoids calling the `SetSession` function when the variables are copied to an initial session. This means that `s.EnableListTablePartition` will only be what the default value was, and does not correctly inherit the global value.

Reply:
Here is the original PR if you want to follow it: https://github.com/pingcap/tidb/pull/24836

Reply:
I removed it, but not sure I completely understood the meaning of `skipInit` since I tried both true and false without affecting the extended test case (i.e. a new session would always be correctly copied from the global variable) but I did not dig deeper than that.